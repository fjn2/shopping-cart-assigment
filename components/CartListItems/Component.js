import Button from "react-bootstrap/Button"
import Col from "react-bootstrap/Col"
import Row from "react-bootstrap/Row"
import ProductImage from "../ProductImage"

const CartListItemsComponent = ({
  cart,
  onItemRemove,
}) => {
  return (
    <div>
      {Object.keys(cart).map((cartKey) => {
        const cartItem = cart[cartKey]
        return (
          <Row>
            <Col>
              <ProductImage  name={`${cartItem.name}_${cartItem.price}`} />
            </Col>
            <Col>
              {cartItem.name}
            </Col>
            <Col>
              {cartItem.brand}
            </Col>
            <Col>
              {cartItem.price} sek
            </Col>
            <Col>
              {Object.keys(cartItem.itemOptions).map((optionKey) => {
                return (
                  <div key={optionKey}>
                    <span style={{ textTransform: 'capitalize' }}>{optionKey}: </span>
                    <b>{cartItem.itemOptions[optionKey]}</b>
                  </div>
                )
              })}
            </Col>
            <Col>
              {cartItem.quantity}
            </Col>
            <Col>
              <Button onClick={() => onItemRemove(cartKey)}>Remove</Button>
            </Col>
          </Row>
        )
      })}
    </div>
  )
}
export default CartListItemsComponent