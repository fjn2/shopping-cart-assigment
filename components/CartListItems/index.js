import useCart from "../../store/cart/hook"
import CartListItemsComponent from "./Component"

const CartListItems = () => {
  const { cart, removeItem } = useCart()

  const onItemRemove = (key) => {
    removeItem(key)
  }

  return <CartListItemsComponent cart={cart} onItemRemove={onItemRemove} />
}
export default CartListItems