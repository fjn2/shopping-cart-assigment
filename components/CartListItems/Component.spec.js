import { shallow } from 'enzyme';
import CartListItems from './Component';

describe('CartListItems', () => {
  it('Should render correctly', () => {
    const props = {
      cart: {
        itemId1: {
          id: 10,
          name: 'Test product',
          brand: 'Test brand name',
          price: 124,
          itemOptions: {
            color: 'red'
          }
        }
      },
      products:[],
      onItemRemove: jest.fn()
    }

    const elem = shallow(<CartListItems {...props} />)
    
    expect(elem).toMatchSnapshot()
  })
  it('Should call `onItemRemove` when the Remove button is clicked', () => {
    const props = {
      cart: {
        itemId1: {
          id: 10,
          name: 'Test product',
          brand: 'Test brand name',
          price: 124,
          itemOptions: {
            color: 'red'
          }
        }
      },
      products: [],
      onItemRemove: jest.fn()
    }

    const elem = shallow(<CartListItems {...props} />)
    
    elem.find('Button').simulate('click')

    expect(props.onItemRemove).toHaveBeenCalledWith('itemId1')
  })
})