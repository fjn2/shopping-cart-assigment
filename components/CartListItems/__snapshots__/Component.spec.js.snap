// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CartListItems Should render correctly 1`] = `
<div>
  <Row>
    <Col>
      <ProductImage
        name="Test product_124"
      />
    </Col>
    <Col>
      Test product
    </Col>
    <Col>
      Test brand name
    </Col>
    <Col>
      124
       sek
    </Col>
    <Col>
      <div
        key="color"
      >
        <span
          style={
            Object {
              "textTransform": "capitalize",
            }
          }
        >
          color
          : 
        </span>
        <b>
          red
        </b>
      </div>
    </Col>
    <Col />
    <Col>
      <Button
        active={false}
        disabled={false}
        onClick={[Function]}
        variant="primary"
      >
        Remove
      </Button>
    </Col>
  </Row>
</div>
`;
